pytest/tests/sanity/catchup_flat_storage_deletions.py

```diff
@@ -0,0 +1,178 @@
+            'secs': 0,
+            'nanos': 500000000
+        },
+        'log_summary_style': 'plain',
+        'state_sync': {
+            'sync': {
+                'ExternalStorage': {
+                    'location': {
+                        'Filesystem': {
+                            'root_dir': state_parts_dir
+                        }
+                    }
+                }
+            }
+        },
+        'state_sync_enabled': True,
+        'consensus.state_sync_timeout': {
+            'secs': 0,
+            'nanos': 500000000
+        },
+        # The schedule means that the node tracks all shards all the time except for epoch heights 2 and 3.
+        # Those epochs correspond to block heights [EPOCH_LENGTH * 2 + 1, EPOCH_LENGTH * 4].
+        'tracked_shard_schedule': [
+            [0],  # epoch_height = 0 and 4
+            [0],  # epoch_height = 1* and 1 and 5
+            [],  # epoch_height = 2
+            [],  # epoch_height = 3
+        ],
+        'tracked_shards': [],
+    }
+
+    config = load_config()
+    nodes = start_cluster(1, 1, 1, config, [["epoch_length", EPOCH_LENGTH]], {
+        0: config0,
+        1: config1
+    })
+    [boot_node, node] = nodes
+
+    logger.info('started the nodes')
+
+    test_account_id = node.signer_key.account_id
+    test_account_key = node.signer_key
+    account_ids = [boot_node.signer_key.account_id, test_account_id]
+
+    print_balances(nodes, account_ids)
+
+    nonce = 10
+
+    latest_block = utils.wait_for_blocks(boot_node,
+                                         target=int(2.5 * EPOCH_LENGTH))
+    epoch = epoch_height(latest_block.height)
+    assert epoch == 2, f"epoch: {epoch}"
+    node.kill()
+    # Restart the node to make it start without opening any flat storages.
+    node.start(boot_node=boot_node)
+    logger.info(f'We are in epoch {epoch}, and the node is restarted')
+
+    print_balances(nodes, account_ids)
+
+    # Delete the account.
+    latest_block_hash = boot_node.get_latest_block().hash_bytes
+    nonce += 1
+    tx = transaction.sign_delete_account_tx(test_account_key, test_account_id,
+                                            boot_node.signer_key.account_id,
+                                            nonce, latest_block_hash)
+    result = boot_node.send_tx(tx)
+    logger.info(result)
+    logger.info(f'Deleted {test_account_id}')
+
+    # Wait until the node tracks the shard and probably does the catchup.
+    latest_block = utils.wait_for_blocks(boot_node,
+                                         target=int(4.5 * EPOCH_LENGTH))
+    epoch = epoch_height(latest_block.height)
+    assert epoch == 4, f"epoch: {epoch}"
+    logger.info(f'We are in epoch {epoch}')
+
+    # Ensure the non-validator node has caught up.
+    utils.wait_for_blocks(node, target=int(4.5 * EPOCH_LENGTH))
+    logger.info(f'The other node is in sync')
+
+    print_balances(nodes, account_ids)
+
+    # Send tokens and expect the transaction to fail.
```

Comment:
Is this comment accurate? It seems like we are checking that the transaction was accepted by the RPC, not that executing it failed. At least I can't find any checks on the transaction outcome or actual balances.

Reply:
adjusted the comment, thanks.